tests/test_model_generator.py

```diff
@@ -137,6 +145,29 @@ def test_type_converter_simple(test_openapi_types, expected_python_types):
                 import_types=["from datetime import datetime"],
             ),
         ),
+        (
+            Schema(type=DataType.STRING, schema_format="date"),
+            TypeConversion(
+                original_type="string",
+                converted_type="date",
+                import_types=["from datetime import date"],
+            ),
+        ),
+        (
+            Schema(type=DataType.STRING, schema_format="decimal"),
+            TypeConversion(
+                original_type="string",
+                converted_type="Decimal",
+                import_types=["from decimal import Decimal"],
+            ),
+        ),
+        (
+            Schema(type=DataType.STRING, schema_format="email"),
```

Comment:
Note that "email" isn't specifically handled by the generator, this is just the test to make sure unexpected formats don't break.